pkg/infra/teams/team3/election_strat.go

```diff
@@ -184,81 +158,49 @@ func (a *AgentThree) calcW2(baseAgent agent.BaseAgent, w2 float64) float64 {
 	return w2
 }
 
-// alg 5
-func (a *AgentThree) CalcReputation(baseAgent agent.BaseAgent) map[commons.ID]float64 {
+func (a *AgentThree) Reputation(baseAgent agent.BaseAgent) map[commons.ID]float64 {
 	view := baseAgent.View()
-	agentState := view.AgentState()
+	vAS := view.AgentState()
+	ids := commons.ImmutableMapKeys(vAS)
 
-	sample := rand.Intn(int(math.Ceil(float64(agentState.Len())*a.sample_percent))-1) + 1
-	counter := 1
-
-	currentLevel := int(view.CurrentLevel())
-	// init  history
-	if currentLevel == a.fightDecisionsHistory.Len()-1 {
-		w1 = 0.0
-		w2 = 0.0
-
-		itr := agentState.Iterator()
-		for !itr.Done() {
-			if counter%sample == 0 {
-				id, _, _ := itr.Next()
-				//idN, _ := strconv.Atoi(id)
-				//fmt.Println(idN)
-				pastHP[id] = GetStartingHP()
-				pastStamina[id] = GetStartingStamina()
-			} else {
-				itr.Next()
-				counter++
-			}
-
-		}
-	}
 	productivity := 5.0
 	needs := 5.0
 	fairness := make(map[commons.ID]float64)
-	sortedfairness := make(map[commons.ID]float64)
 
-	itr := agentState.Iterator()
-	counter1 := 1
+	// Number of agents to sample for KA (fixed)
+	sampleLength := int(math.Max(float64(a.numAgents)*a.sample_percent, float64(vAS.Len())))
```

Comment:
Extract right hand side into variables (for checking)

Reply:
also, should we be using the Max function here? I think it should be Min()